### PR TITLE
change the type of ports form PortBinding to TransportPort in link.go

### PR DESCRIPTION
**Pull request by fmzhen**
I think the netutils.TransportPort is better  than netutils.PortBinding.

In the Docker link operation. We just need the child container's ExposedPorts.

@mavenugo  @aboch @mrjana  @resouer
Signed-off-by: Mingzhen Feng <fmzhen@zju.edu.cn>

### CODE REVIEW

**Comment by aboch** [comment]
You need to also change bridge.go where newLink() is being called.

**Comment by fmzhen** [comment]
@aboch  @mavenugo @mrjana

**Comment by aboch** [comment]
@fmzhen, Your change is not passing a test. If you can take care of it very quickly it would be great, otherwise I am going ahead and make the changes in libnetwork and options pakcage and will wait for you to rebase and merge your changes in bridge.go.

Sorry for the rush, but this is a blocker for libnetwork integration in docker. Worst case, I will commit your changes myself and put you as Author.

File: drivers/bridge/bridge.go [hunk 1]
```diff
@@ -754,6 +755,14 @@ func parseEndpointOptions(epOptions map[string]interface{}) (*EndpointConfigurat
 		}
 	}
 
+	if opt, ok := epOptions[options.ExposedPorts]; ok {
```
Comment (aboch) [inline drivers/bridge/bridge.go]: You did not take care of the test code. How can the test pass if it was feeding portmapping for link/unlink tests but now driver discards that ?

**Comment by aboch** [comment]
@fmzhen, please look at https://github.com/docker/libnetwork/pull/119. 
I took care of the missing changes in test code and added the changes needed in endpoint.go, working on top of your commit.

So now you can close this PR, your changes are part of the above PR.

Thanks for your contribution !

**Comment by mavenugo** [comment]
@aboch thanks for merging @fmzhen's changes via #119. Closing it.

**Pull request closed without merging**